starknet_devnet/transactions.py

```diff
@@ -194,8 +195,13 @@ def __get_transaction_by_hash(self, tx_hash: str) -> DevnetTransaction or None:
         """
         Get a transaction by hash.
         """
-        numeric_hash = int(tx_hash, 16)
-        return self.__instances.get(numeric_hash)
+        if tx_hash.startswith("0x"):
```

Comment:
String starting with `0x` doesn't necessarily make it a hex string. It should be wrapped in a try-except block to prevent Internal Server Errors caused by strings like `0xyz`. E.g.:

```python
if tx_hash.startswith("0x"):
    try:
        return int(tx_id, 16)
    except ValueError:
        pass

raise ...
```

Reply:
right, I'll update tests and make a fix

Reply:
Fixed in 12a619b372e4cb10a44fa01ad3566430ea5ba5b5